go.mod

```diff
@@ -1,96 +1,97 @@
 module github.com/rhobs/obsctl-reloader
 
-go 1.17
-
-replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v1.8.2-0.20210621150501-ff58416a0b02
+go 1.18
```

Comment:
This is a restriction, not an upgrade of the Go version used to build the project. Is there any feature we are using that is available only in Go 1.18?

Reply:
IIRC, we chose go 1.17 to align with obsctl? not sure what is the question if i'm being honest :sweat_smile:

Reply:
Sorry for not being clear! I was trying to upgrade the reloader to latest Go version while upgrading other modules, and found the ubi8/go-toolset only goes up to go 1.18. The question was if there are any other builder images that you'd recommend that support go 1.19/1.20?

@periklis recommended something like https://github.com/openshift/elasticsearch-operator/blob/release-5.6/Dockerfile#L3

Reply:
@maorfr do you know when we will have `ubi8/go-toolset` images for Go 1.19 and 1.20? Ideally we want to always be compiling with the latest possible version of Go to take advantages of compiler fixes and improvements, particularly those that can improve GC time, perf, etc.

I mention why not move to Go 1.19 straight away, but there is already 1.20. It would be better to go straight to the latest.

Reply:
i have no information on the release process or cadence of this image unfortunately

Reply:
We can also use `golang:1.19.7` as builder image like the one used in loki operator https://github.com/grafana/loki/blob/9844fad8b4110a1a706b114335402228599c81d9/operator/Dockerfile#L2

Reply:
nit: you can't use dockerhub
https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/moving-off-of-docker-hub.md

Reply:
UBI8 is usually lagging behind and it might take too-long to wait for it. That's the main reason of the openshift-golang-builder images upstream.

Reply:
We support more than these two at RedHat ;)

Reply:
That's cool. But it won't be helpful if the tools we are working with (linters, for example, among others) start to break on Go 1.18.

Reply:
ACK